### DISCUSSION TITLE
Using validationSchema with Yup and defaultValues

### DISCUSSION THREAD

**Discussion by eliw00d** [react-hook-form/react-hook-form#1936] [Q&A] [answered]
So, yup has `mixed.default(value: any): Schema` which allows you to set default values. When using validationSchema are those ignored?

**Comment by bluebill1049** [comment]
`defaultValues` should be at `useForm` for input inline.

**Reply by djMax** [reply]
(why? Couldn't yup be the source for this to simplify setup and portability?)

**Reply by bluebill1049** [reply]
we don't only support yup at moment, but feel free to send a PR at resolver. happy to take a look at it.

**Comment by terebentina** [comment, marked as answer]
yup has `schema.cast()` which can be used for the defaultValues of useForm. I believe this would solve your problem, @eliw00d 
Something like this:
```
  const methods = useForm({
    defaultValues: validationSchema.cast(),
    resolver: yupResolver(validationSchema, { stripUnknown: true, abortEarly: false }),
  });
```

**Reply by eliw00d** [reply]
This is great! Thank you for that!

**Reply by bluebill1049** [reply]
ty @terandrei 🙏

**Reply by bilsak18** [reply]
I am really needing this. Thank you for it. 

Concerning the cast method, I see that it requires arguments and I don't know what arguments to pass in order to achieve the same thing in your comment.

**Reply by tomocrafter** [reply]
@bilsak18 Maybe you can try `getDefault()` which returns default value of schema as you can tell.

**Reply by terebentina** [reply]
You don't need to pass anything to `.cast()` if all you need is the default values. Either `.cast()` or `.cast({})` will work and do the same thing.
